airbot_sensor_interface_node:
  ros__parameters:

    target_frame: "base_link" # "map" or "base_link"

    tof:
      all:
        use: true
      1D:
        use: true
        publish_rate_ms: 10
        tilting_angle_deg: 45.0 # double type
      multi:
        publish_rate_ms: 50
        left:
          use: true
          pitch_angle_deg: -0.0 # (-)가 들리는 방향
        right:
          use: true
          pitch_angle_deg: -0.0 # (-)가 들리는 방향
        row:
          use: true
          publish_rate_ms: 50

    camera:
      use: true
      publish_rate_ms: 100
      pointcloud_resolution: 0.05
      class_id_confidence_th: # 형식: "class id: confidence score"
        - "0: 50" #cable
        - "1: 75" #carpet
        - "2: 60" #clothes
        # - "5: 80" #obstacle 미사용
        - "6: 30" #poop
        # - "8: 80" #threshold 미사용
      object_direction: false  # 정방향(CCW+):True, 역방향(CW+):False
      object_max_distance_m: 3.0 # 최대 장애물 거리 (해당 거리 이내의 장애물 정보만 처리)
      logger:
        use: true
        margin:
          distance_diff: 0.5
          width_diff: 0.1
          height_diff: 0.1

    cliff:
      use: true
      publish_rate_ms: 10

    collision:
      use: true
      publish_rate_ms: 10

    use_sim_time: false

################################
#######  ToF Matrix idx  #######
#                              #
#       [00, 01, 02, 03]       #
#       [04, 05, 06, 07]       #
#       [08, 09, 10, 11]       #
#       [12, 13, 14, 15]       #
#                              #
################################
#######  Camera Object  ########
######     Class ID      #######
#                              #
#        0: cable              #
#        1: carpet             #
#        2: clothes            #
#        3: liquid             #
#        4: non_obstacle       #
#        5: obstacle           #
#        6: poop               #
#        7: scale              #
#        8: threshold          #
#        9: person             #
#        10: dog               #
#        11: cat               #
#                              #
################################
###########  Bot IR  ###########
######### < Top View > #########
#                              #
#              ff              #
#          fl      fr          #
#                              #
#          bl      br          #
#              bb              #
#                              #
################################